cartographer_ros/src/cartographer_node_main.cc

```diff
@@ -90,6 +93,20 @@ Rigid3d ToRigid3d(const geometry_msgs::TransformStamped& transform) {
                                     transform.transform.rotation.z));
 }
 
+Rigid3d ToRigid3d(const geometry_msgs::Pose& pose) {
+  return Rigid3d(Eigen::Vector3d(pose.position.x,
+                                 pose.position.y,
+                                 pose.position.z),
+                 Eigen::Quaterniond(pose.orientation.w,
+                                    pose.orientation.x,
+                                    pose.orientation.y,
+                                    pose.orientation.z));
+}
+
+PoseCovariance ToPoseCovariance(boost::array<double, 36ul> covariance) {
+  return Eigen::Map<Eigen::Matrix<double,6,6,Eigen::RowMajor> > (covariance.data());
```

Comment:
Covariance matrices are symmetric, so there is no need to specify RowMajor. Remove?